coral-schema/src/main/java/com/linkedin/coral/schema/avro/ViewToAvroSchemaConverter.java

```diff
@@ -92,6 +94,19 @@ public Schema toAvroSchema(String dbName, String tableOrViewName) {
     return avroSchema;
   }
 
+  /**
+   * An API to generate the avro schema for a SQL string
+   *
+   * @param sql SQL string literal
+   * @return avro schema for a given sql query
+   */
+  public Schema toAvroSchema(String sql) {
+    Preconditions.checkNotNull(sql);
+
+    RelNode relNode = hiveToRelConverter.convertSql(sql);
+    return relToAvroSchemaConverter.convert(relNode, false);
+  }
```

Comment:
I'm not sure about the idea to expose a public method just for testing. Could you make it only visible to test module?

Reply:
I actually think this method is a good public API to be exposed, as does the `convertSql` is public in `HiveToRelConverter`.

Reply:
Discussed offline. We need to think more about the API design and do not release it as public API for now.